Mypy 1.12 introduces return type error in sympy/core/exprtools.py

### Issue: Mypy 1.12 introduces a type error in `sympy/core/exprtools.py`

#### Description:
When running mypy version 1.12 on the SymPy codebase, a type error is encountered, which did not occur with the previous version (mypy-1.11.2). The error points to an issue in `sympy/core/exprtools.py` regarding an incompatible return value type.

#### Steps to Reproduce:
1. Run the following command with mypy version 1.12:
   ```bash
   mypy sympy

The following error occurs:
```
sympy/core/exprtools.py:281: error: Incompatible return value type (got "Any | tuple[Expr, int]", expected "tuple[Expr, Rational]")  [return-value]
Found 1 error in 1 file (checked 1509 source files)
Error: Process completed with exit code 1.

```

--- a/sympy/core/exprtools.py
+++ b/sympy/core/exprtools.py
@@ -277,8 +277,11 @@ def decompose_power_rat(expr: Expr) -> tTuple[Expr, Rational]:
     (exp(x/2), -3)
 
     """
-    _ = base, exp = expr.as_base_exp()
-    return _ if exp.is_Rational else decompose_power(expr)
+    base, exp = expr.as_base_exp()
+    if not exp.is_Rational:
+        base, exp_i = decompose_power(expr)
+        exp = Integer(exp_i)
+    return base, exp # type: ignore
 
 
 class Factors:

